### PR TITLE
docs(body): add example for to_bytes()

### PR DESCRIPTION
**Pull request by Vagelis-Prokopiou**
cc #2201

### CODE REVIEW

**Comment by Vagelis-Prokopiou** [comment]
Hi dear Sean.

I don't know if this PR helps. I decided to give it a try anyway :-)

File: src/body/to_bytes.rs [hunk 1]
```diff
@@ -13,6 +13,28 @@ use super::HttpBody;
...
+/// ```
+/// let response = client.request(req).await?;
+///
+/// const MAX_ALLOWED_RESPONSE_SIZE: u32 = 1024;
+///
+/// let response_content_length: u32 = response.headers()
+///     .get(hyper::header::CONTENT_LENGTH)
+///     .expect("Failed to extract the CONTENT_LENGTH")
+///     .to_str().expect("Failed to_str() the CONTENT_LENGTH")
+///     .parse::<u32>().expect("Failed to parse the CONTENT_LENGTH");
```
Comment (seanmonstar) [inline src/body/to_bytes.rs:30]: It does seem like a good idea to remind people in the example to care about size, so thanks! It should be possible to more easily get the reported size from the body itself, after importing `hyper::body::HttpBody`. Then you can check `response.body().size_hint()`.

File: src/body/to_bytes.rs [hunk 2]
```diff
@@ -13,6 +13,28 @@ use super::HttpBody;
 /// Care needs to be taken if the remote is untrusted. The function doesn't implement any length
 /// checks and an malicious peer might make it consume arbitrary amounts of memory. Checking the
 /// `Content-Length` is a possibility, but it is not strictly mandated to be present.
+///
+/// Create an empty `Body` stream.
+///
+/// # Example
+///
+/// ```
+/// let response = client.request(req).await?;
```
Comment (seanmonstar) [inline src/body/to_bytes.rs:22]: You'll need to put a couple "hidden" lines above to allow the usage of `.await` and `?`. Something like:

```
/// # async fn ex() -> Result<(), Box<dyn std::error::Error>> {
```

And then close the function in another hidden line at the bottom.

File: src/body/to_bytes.rs [hunk 3]
```diff
@@ -13,6 +13,28 @@ use super::HttpBody;
...
+/// const MAX_ALLOWED_RESPONSE_SIZE: u32 = 1024;
+///
+/// let response_content_length: u32 = response.headers()
+///     .get(hyper::header::CONTENT_LENGTH)
+///     .expect("Failed to extract the CONTENT_LENGTH")
+///     .to_str().expect("Failed to_str() the CONTENT_LENGTH")
+///     .parse::<u32>().expect("Failed to parse the CONTENT_LENGTH");
+///
+/// if response_content_length < MAX_ALLOWED_RESPONSE_SIZE {
+///     let body_bytes = hyper::body::to_bytes(response.into_body()).await.expect("body::to_bytes failed");
```
Comment (seanmonstar) [inline src/body/to_bytes.rs:33]: Since using `?` already, it'd be less noisy to keep using it and not need to have large `.expect(reason)`.

**Review by seanmonstar — CHANGES_REQUESTED** [review]
Thanks for the PR! I left some suggestions inline.

File: src/body/to_bytes.rs (hunk 1 above)
Comment (Vagelis-Prokopiou) [reply src/body/to_bytes.rs]: Addressed with commit d98dbfc9

Comment (Vagelis-Prokopiou) [reply src/body/to_bytes.rs]: Addressed with commit d98dbfc9

**Comment by Vagelis-Prokopiou** [comment]
Hi Sean :-)

The current state is the closest that I managed to reach executing the doc function. It is failing, but I did not manage to resolve the error no matter what I tried.

The current error is the following:
```
---- src/body/to_bytes.rs - body::to_bytes::to_bytes (line 21) stdout ----
error[E0433]: failed to resolve: could not find `Client` in `hyper`
  --> src/body/to_bytes.rs:30:21
     |
12 | let client = hyper::Client::new();
     |                     ^^^^^^ could not find `Client` in `hyper`
```

I tried to resolve it by adding `# #[cfg(feature = "full")]` or `# #[cfg(feature = "client")]`, bit I did not succeed.

Either I need some guidance, or you can commit directly the changes need. It is your call :-)

File: src/body/to_bytes.rs [hunk 4]
```diff
@@ -13,6 +13,37 @@ use super::HttpBody;
...
+///
+/// let response_content_length = match response.body().size_hint().upper() {
+///     Some(v) => v,
+///     None => MAX_ALLOWED_RESPONSE_SIZE + 1 // Just to protect ourselves from a malicious response
+/// };
+///
+/// if response_content_length < MAX_ALLOWED_RESPONSE_SIZE {
+///     let body_bytes = hyper::body::to_bytes(response.into_body()).await?;
+///     let body_string = String::from_utf8(body_bytes.to_vec()).expect("!String::from_utf8");
+///     println!("body: {}", body_string);
```
Comment (seanmonstar) [inline src/body/to_bytes.rs:41]:
```suggestion
///     println!("body: {:?}", body_bytes);
```

File: src/body/to_bytes.rs [hunk 5]
```diff
@@ -13,6 +13,37 @@ use super::HttpBody;
...
+/// const MAX_ALLOWED_RESPONSE_SIZE: u64 = 1024;
+///
+/// let response_content_length = match response.body().size_hint().upper() {
+///     Some(v) => v,
+///     None => MAX_ALLOWED_RESPONSE_SIZE + 1 // Just to protect ourselves from a malicious response
+/// };
+///
+/// if response_content_length < MAX_ALLOWED_RESPONSE_SIZE {
+///     let body_bytes = hyper::body::to_bytes(response.into_body()).await?;
+///     let body_string = String::from_utf8(body_bytes.to_vec()).expect("!String::from_utf8");
```
Comment (seanmonstar) [inline src/body/to_bytes.rs:40]:
```suggestion
```

**Review by seanmonstar — APPROVED** [review]
Nice, thanks for the fixes! Sorry for the delay, I'm going to suggest a couple small changes and force them through, and then merge.

**Comment by Vagelis-Prokopiou** [comment]
@seanmonstar thanx for helping/merging this. Yet, as I stated in my [previous comment](https://github.com/hyperium/hyper/pull/2487#issuecomment-816823291), the doc tests fail. Running the doc tests locally for master has the following result:
```
failures:

---- src/body/to_bytes.rs - body::to_bytes::to_bytes (line 19) stdout ----
error[E0433]: failed to resolve: could not find `Client` in `hyper`
  --> src/body/to_bytes.rs:28:21
   |
12 | let client = hyper::Client::new();
   |                     ^^^^^^ could not find `Client` in `hyper`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
Couldn't compile the test.

failures:
    src/body/to_bytes.rs - body::to_bytes::to_bytes (line 19)

test result: FAILED. 3 passed; 1 failed; 2 ignored; 0 measured; 0 filtered out; finished in 0.25s

error: test failed, to rerun pass '--doc'
```
Doesn't it matter? Can't it be resolved?

**Pull request merged**